File: src/main/java/com/github/pagviewer/nativebridge/PagNativeLibrary.java
package com.github.pagviewer.nativebridge;

import java.nio.ByteBuffer;

public interface PagNativeLibrary {
    int COLOR_TYPE_BGRA_8888 = 3;
    int ALPHA_TYPE_UNPREMULTIPLIED = 3;

    long loadFile(byte[] bytes, String filePath);

    long createDecoder(long composition, float maxFrameRate, float scale);

    int compositionWidth(long composition);

    int compositionHeight(long composition);

    int width(long decoder);

    int height(long decoder);

    int frameCount(long decoder);

    float frameRate(long decoder);

    boolean readFrame(long decoder, int frameIndex, ByteBuffer destination, int rowBytes);

    void release(long handle);
}

File: src/main/kotlin/com/github/pagviewer/nativebridge/PagNativeLibrary.kt
package com.github.pagviewer.nativebridge

import java.nio.ByteBuffer

interface PagNativeLibrary {
    fun loadFile(bytes: ByteArray, filePath: String): Long

    fun createDecoder(composition: Long, maxFrameRate: Float, scale: Float): Long

    fun compositionWidth(composition: Long): Int

    fun compositionHeight(composition: Long): Int

    fun width(decoder: Long): Int

    fun height(decoder: Long): Int

    fun frameCount(decoder: Long): Int

    fun frameRate(decoder: Long): Float

    fun checkFrameChanged(decoder: Long, frameIndex: Int): Boolean

    fun readFrame(decoder: Long, frameIndex: Int, destination: ByteBuffer, rowBytes: Int): Boolean

    fun release(handle: Long)

    companion object {
        const val COLOR_TYPE_BGRA_8888 = 3
        const val ALPHA_TYPE_UNPREMULTIPLIED = 3
    }
}
